### PR TITLE
fix: Update ProgramRuleActionType values in code and DB [DHIS2-7426]

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

**Comment by enricocolasante** [comment]
The build needs https://github.com/dhis2/dhis2-rule-engine/pull/19 to be merged

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java [hunk 1]
```diff
@@ -40,17 +40,15 @@
  */
 public enum ProgramRuleActionType
 {
-    DISPLAYTEXT( "displaytext" ),
-    DISPLAYKEYVALUEPAIR( "displaykeyvaluepair" ),
```
Comment (Bekkalizer) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java]: DisplayKeyValuePair is different from DisplayText, so they cant be replaced with showmessage directly without losing functionality.

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.34/V2_34_3__Update_program_rule_actions_and_update_evaluation_time.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+update programruleaction set actiontype = 'SHOWMESSAGE' where actiontype = 'DISPLAYKEYVALUEPAIR' OR actiontype = 'DISPLAYTEXT';
+update programruleaction set actiontype = 'SHOWWARNING', evaluationtime = 'ON_COMPLETE' where actiontype = 'WARNINGONCOMPLETE';
+update programruleaction set actiontype = 'SHOWERROR', evaluationtime = 'ON_COMPLETE' where actiontype = 'ERRORONCOMPLETE';
```
Comment (Bekkalizer) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.34/V2_34_3__Update_program_rule_actions_and_update_evaluation_time.sql]: Assuming we must also set the WHEN parameter properly for these(on complete)

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java]: I was following the WHEN table in https://jira.dhis2.org/browse/DHIS2-7425

Should we change that to not group these 2 together into SHOWMESSAGE?

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.34/V2_34_3__Update_program_rule_actions_and_update_evaluation_time.sql (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.34/V2_34_3__Update_program_rule_actions_and_update_evaluation_time.sql]: I am doing it.
I am changing the actionType and the evaluationTime

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java (hunk 1 above)
Comment (Bekkalizer) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleActionType.java]: Ok, thanks for pointing that out. We should not group them. Added a comment in the Jira to have Rodolfo update the spec.

**Pull request closed without merging**